tests/test_templates.py

```diff
@@ -0,0 +1,41 @@
+"""
+test_custom_extension_in_hooks.
+
+Tests to ensure custom cookiecutter extensions are properly made available to
+pre- and post-gen hooks.
+"""
+import codecs
+import os
+
+import pytest
+
+from cookiecutter import main
+
+
+@pytest.fixture
+def output_dir(tmpdir):
+    """Fixture. Create and return custom temp directory for test."""
+    return str(tmpdir.mkdir('templates'))
+
+
+@pytest.mark.parametrize("template", ["include", "no-templates", "extends", "super"])
+def test_build_templates(template, output_dir):
+    """
+    Verify Templates Design keywords.
+
+    no-templates is a compatibility tests for repo without `templates` directory
```

Comment:
Typo: `tests` should be `test`.

Reply:
> One nitpick.
> 
> Was there no other way to handle this than forking the base repo?
> 
> LGMT otherwise.
> 
> @abvikg you write some really clean code. I enjoy reading it.

This commit was authored by https://github.com/simobasso/ :)

I am essentially duplicating his PR to the original cookiecutter repo here https://github.com/cookiecutter/cookiecutter/pull/1485.